src/CPUCore.cpp

```diff
@@ -295,12 +295,18 @@ static bool verifyJITSizeOf(std::string structname, etiss::int32 expected_size,
         return false;
     // generate code
     std::string error;
-    std::string code = std::string(prefix + "\n#include \"etiss/jit/CPU.h\"\n#include "
-                                            "\"etiss/jit/System.h\"\n#include \"etiss/jit/ReturnCode.h\"\n "
-                                            "#include \"etiss/jit/types.h\"\n#include "
-                                            "\"etiss/jit/fpu/softfloat.h\"\n etiss_int32 get_size(){ return "
-                                            "sizeof(") +
-                       structname + ");}";
+    std::string code = std::string(prefix + R"V0G0N(
+#include "etiss/jit/Coverage.h"
+#include "etiss/jit/libCSRCounters.h"
+#include "etiss/jit/libresources.h"
+#include "etiss/jit/libsemihost.h"
+#include "etiss/jit/libsoftfloat.h"
```

Comment:
Not a fan of having riscv-specific includes in here. This is something that should be added by the arch…

Isn’t this what `jit.external_headers` is supposed to do? https://github.com/tum-ei-eda/etiss/blob/7c98f11587336dc63b4cc77789c45c9031a33b6d/src/ETISS.cpp#L741

I am fine with the fix because it is better to have it like this than fully broken.

Reply:
CSRs (mtime, minstret, mcycle) are RISCV-only.

Softfloat might only be used by a core implementing FP instructions. 

But we can keep the includes as they are, my comment was just a minor option.